src/wstool/multiproject_cli.py

```diff
@@ -621,6 +622,99 @@ def cmd_diff(self, target_path, argv, config=None):
 
         return False
 
+    def cmd_foreach(self, target_path, argv, config=None):
+        """Run shell commands in each repository."""
+        parser = OptionParser(
+            usage='usage: %s foreach [localname]* [command] [OPTIONS]' % self.progname,
```

Comment:
so if both vcsfileter `--(git|hg|bzr|svn)` and localnames are provided, raise Warn?